src/adapters/mc/CMakeLists.txt

```diff
@@ -1,6 +1,159 @@
+        COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:${ARG_TARGET_NAME}> ${PROJECT_BINARY_DIR}/GetStaging/Modules/DscNativeResources/OsConfigResource/libOsConfigResource.so
+    )
+    set(STAGING_COMMANDS_SET
+        COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/SetStaging
+        COMMAND ${CMAKE_COMMAND} -E copy_if_different "${ARG_META_CONFIG_JSON_FILE}" "${PROJECT_BINARY_DIR}/SetStaging/Set${ARG_DESTINATION_FILE}.metaconfig.json"
+        COMMAND ${CMAKE_COMMAND} -E copy_if_different "${ARG_MOF_FILE}" "${PROJECT_BINARY_DIR}/SetStaging/Set${ARG_DESTINATION_FILE}.mof"
+        COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:${ARG_TARGET_NAME}> ${PROJECT_BINARY_DIR}/SetStaging/Modules/DscNativeResources/OsConfigResource/libOsConfigResource.so
+    )
+
+    # Add telemetry binary if enabled
+    if(ENABLE_TELEMETRY)
+        list(APPEND STAGING_COMMANDS_GET
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different "${BINARY_DIR}/exe/telemetry" ${PROJECT_BINARY_DIR}/GetStaging/Modules/DscNativeResources/OsConfigResource/telemetry
+        )
+        list(APPEND STAGING_COMMANDS_SET
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different "${BINARY_DIR}/exe/telemetry" ${PROJECT_BINARY_DIR}/SetStaging/Modules/DscNativeResources/OsConfigResource/telemetry
+        )
+    endif()
+
+    # Add additional commands if provided
+    if(ARG_ADDITIONAL_COMMANDS_GET)
+        list(APPEND STAGING_COMMANDS_GET ${ARG_ADDITIONAL_COMMANDS_GET})
+    endif()
+    if(ARG_ADDITIONAL_COMMANDS_SET)
+        list(APPEND STAGING_COMMANDS_SET ${ARG_ADDITIONAL_COMMANDS_SET})
+    endif()
+
+    # Build dependencies list
+    set(STAGING_DEPENDS ${ARG_TARGET_NAME} ${ARG_ADDITIONAL_DEPENDS})
+    if(ENABLE_TELEMETRY)
+        list(APPEND STAGING_DEPENDS telemetry)
+    endif()
+
+    add_custom_target(stage_create_get_${ARG_TARGET_NAME}_zip
+        ${STAGING_COMMANDS_GET}
+        DEPENDS ${STAGING_DEPENDS})
+
+    add_custom_target(create_get_${ARG_TARGET_NAME}_zip ALL
+        BYPRODUCTS ${OsConfigRootBinaryDir}/${ARG_DESTINATION_FILE}.zip
+        COMMAND ${CMAKE_COMMAND} -E tar "cfv" "${OsConfigRootBinaryDir}/${ARG_DESTINATION_FILE}.zip" --format=zip .
+        DEPENDS stage_create_get_${ARG_TARGET_NAME}_zip
+        WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/GetStaging/)
+
+    add_custom_target(stage_create_set_${ARG_TARGET_NAME}_zip
+        ${STAGING_COMMANDS_SET}
+        DEPENDS ${STAGING_DEPENDS})
+
+    add_custom_target(create_${ARG_TARGET_NAME}_zip ALL
+        BYPRODUCTS ${OsConfigRootBinaryDir}/${ARG_DESTINATION_FILE}.zip
+        COMMAND ${CMAKE_COMMAND} -E tar "cfv" "${OsConfigRootBinaryDir}/Set${ARG_DESTINATION_FILE}.zip" --format=zip .
+        DEPENDS stage_create_set_${ARG_TARGET_NAME}_zip
+        WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/SetStaging/)
+endfunction()
+
 add_compile_options("-Wall;-Wextra;-Wunused;-Werror;-Wformat;-Wformat-security;-Wno-unused-result")
```

Comment:
How is this simpler? Is not. Also, there is a dual purpose of the fragments removed from each policy CMakeList: 

1. The staged files.
2. The ZIP.

... and not just the ZIP.

Suggest to put back what was in the CMakeList.txt for ASB and SSH and remove this uber-complication from here.

Lastly, why is not done in a separate PR? It's completely unrelated to everything else. 

Please split in separate PRs.

Reply:
Correct and we are doing both those things, staging into a separate directory AND creating the ZIP.

I will do the change as a separate PR to remove some of the confusion, but this simplifies all of that duplicate code for staging/creating the zip packages significantly